jvm-packages/xgboost4j-spark-gpu/src/test/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuPerTest.scala

```diff
@@ -0,0 +1,292 @@
+/*
+ Copyright (c) 2021 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala.rapids.spark
+
+import java.nio.file.{Files, Path}
+import java.sql.{Date, Timestamp}
+import java.util.{Locale, TimeZone}
+
+import com.nvidia.spark.rapids.RapidsConf
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.network.util.JavaUtils
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.rapids.execution.TrampolineUtil
+
+trait GpuTestSuite extends FunSuite with TmpFolderSuite {
+  import SparkSessionHolder.withSparkSession
+
+  protected def getResourcePath(resource: String): String = {
+    require(resource.startsWith("/"), "resource must start with /")
+    getClass.getResource(resource).getPath
+  }
+
+  def enableCsvConf(): SparkConf = {
+    new SparkConf()
+      .set(RapidsConf.ENABLE_READ_CSV_DATES.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_BYTES.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_SHORTS.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_INTEGERS.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_LONGS.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_FLOATS.key, "true")
+      .set(RapidsConf.ENABLE_READ_CSV_DOUBLES.key, "true")
+  }
+
+  def withGpuSparkSession[U](conf: SparkConf = new SparkConf())(f: SparkSession => U): U = {
+    val c = conf.clone()
+      .set("spark.rapids.sql.enabled", "true")
+//      .set("spark.rapids.sql.explain", "ALL")
```

Comment:
Remove this.

Reply:
It's ok to leave it here. the configuration can explain which spark operators can be replaced by GPU. Sometimes, we need it when debugging.

Reply:
Please use comments for code explanation.  We generally don't include debug code.

Reply:
Done